roles/openshift_storage_glusterfs/tasks/glusterfs_common.yml

```diff
@@ -48,6 +48,30 @@
   - glusterfs_heketi_deploy_is_missing
   - glusterfs_heketi_is_missing
 
+# We wait for pod here because if heketi_deploy_part1 is skipped, our registerd
+# value for deploy_heketi_pod becomes skipped as well.
+- name: Wait for deploy-heketi pod
+  oc_obj:
+    namespace: "{{ glusterfs_namespace }}"
+    kind: pod
+    state: list
+    selector: "glusterfs=deploy-heketi-{{ glusterfs_name }}-pod"
+  register: deploy_heketi_pod_wait
+  until:
+  - "deploy_heketi_pod_wait.results.results[0]['items'] | count > 0"
+  # Pod's 'Ready' status must be True
+  - "deploy_heketi_pod_wait.results.results[0]['items'] | lib_utils_oo_collect(attribute='status.conditions') | lib_utils_oo_collect(attribute='status', filters={'type': 'Ready'}) | map('bool') | select | list | count == 1"
+  delay: 10
+  retries: "{{ (glusterfs_timeout | int / 10) | int }}"
+  when:
+  - glusterfs_heketi_is_native
```

Comment:
@jarrpa I think we should add 'glusterfs_heketi_is_missing' to this boolean list.

Reply:
This does not address my previous comment. :)